File: GFG/Stacks/llimp.py
import ds

class Mstack:
    def __init__(self):
        self.lls=ds.LinkedList()
        self.sz=0
        
    def push(self,data):
        temp=ds.Node(data)
        temp.next=self.lls.head
        self.lls.head=temp
        self.sz+=1
        
    def pop(self):
        res=self.lls.head.data
        self.lls.head=self.lls.head.next
        self.sz-=1
        return res
    
    def top(self):
        return self.lls.head.data
    def isEmpty(self):
        return self.sz==0
    def size(self):
        return self.sz
    
ms=Mstack()
for i in range(1,15):
    ms.push(i*5)
print(ms.size()," ",ms.isEmpty())
count=0
while(ms.isEmpty()==False and count<25):
    print(ms.pop(),end="* ")
    count-=1
    
        
        

File: GFG/Stacks/llimp.cpp
#include<iostream>
#include"../ds.h"
using namespace std;

template <typename T>
class MStack{
    public:
    LinkedList<T> st;
    int sz=0;

    void push(T value){
        Node<T>* temp=new Node(value);
        temp->next=st.head;
        st.head=temp;
        sz++;
    }
    T pop(){
        T res=st.head->data;
        Node<T>* temp=st.head;
        st.head=st.head->next;
        delete temp;
        sz--;
        return res;
    }
    T top(){
        return st.head->data;
    }
    bool isEmpty(){
        return sz==0;
    }
    int size(){
        return sz;
    }
};

int main(){
    MStack<int> st;
    for (int i=1;i<11;i++){
        st.push(i*10);
    }
    cout<<st.size()<<" ** "<<st.isEmpty()<<endl;
    while(st.isEmpty()==false){
        cout<<st.pop()<<endl;
    }
    return 0;
}